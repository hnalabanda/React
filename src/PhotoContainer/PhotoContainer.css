
.photocontainer{
    justify-content: center;
    display:flex;
    flex-direction: row;
    flex-wrap: wrap;
}

.sizeclass{
    margin-left: 1em ;
    margin-bottom: 1em ;
    height:15em;
    width:15em;

}