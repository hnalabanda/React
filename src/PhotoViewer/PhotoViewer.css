.ViewerSize{
    height: 50vh;
  
}

.ViewerContainer{
    text-align: center;
}