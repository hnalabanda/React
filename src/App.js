import React, {useState} from 'react';
import logo from './logo.svg';
import './App.css';
import {PhotoViewer} from "./PhotoViewer/PhotoViewer";
import {ImageUrls} from "./data";
import {PhotoContainer} from "./PhotoContainer/PhotoContainer";


function App() {
    // const images = [];
    //
    //
    // for (let i = 0; i < ImageUrls.length; i++){
    //     images.push(<PhotoViewer imgUrl={ImageUrls[i]}/>)
    // 
    const [selectedImage, setSelectedImage] = useState(ImageUrls[0]);
  
    
    return (
        <div >
            <h1>React Photo Viewer</h1>
            <PhotoViewer src={selectedImage}/>
         <PhotoContainer setSelectedImage={setSelectedImage}/>
        </div>
    );
}


export default App;
