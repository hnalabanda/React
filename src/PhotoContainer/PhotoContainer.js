import React, {useState} from "react";
import {ImageUrls} from "../data";
import './PhotoContainer.css';
import {PhotoViewer} from "../PhotoViewer/PhotoViewer";
//import {imagesArray} from "../App";
export function PhotoContainer(props) {
    const images = ImageUrls.map(url => <img className="sizeclass" src={url} alt="" onClick={() => props.setSelectedImage(url)}/>);
    return (
        
        <div>
       
            <div className="photocontainer"><br/>
                {images} 
            </div>
        </div>
    );
}

