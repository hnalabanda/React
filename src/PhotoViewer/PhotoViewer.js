import React,{useState} from "react";
import './PhotoViewer.css';

function PhotoViewer(props){
    return (
        <div className="ViewerContainer">
            <img src="https://i.picsum.photos/id/360/200/300.jpg"  className="ViewerSize"></img>
        </div>
    );
}

export {PhotoViewer};

