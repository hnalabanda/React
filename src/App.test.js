import React from 'react';
import {render, fireEvent} from '@testing-library/react';
//import renderer from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import {ImageUrls} from "./data";
import {PhotoContainer} from "./PhotoContainer/PhotoContainer";
import {PhotoViewer} from "./PhotoViewer/PhotoViewer";
import {wait} from "@testing-library/dom";

test('renders the correct first link', () => {
 
  const firstURL=ImageUrls[0];

   expect(firstURL).toBe("https://i.picsum.photos/id/600/1600/900.jpg");
});


test('does not include broken images', () => {
   
    const brokenImages = [
        1, 24, 32, 36, 44, 47
    ];
    var brokenLinks=false;
    var imageNumberString;
    
  for(let i=0;i<brokenImages.length;i++){
      imageNumberString = `6${brokenImages[i].toString().padStart(2, '0')}`;
     
      for(let c=0;c<ImageUrls.length;c++){
        
          if(ImageUrls[c].includes(imageNumberString)) {
              brokenLinks=true;
              break;
          }
     
      }
      if(brokenLinks) {
          break;
      }
  }
   

    expect(brokenLinks).toBe(false);
});




    describe("selected image", () => {
    it("The photoviewer component has not changed", async () => {
        const tree  = renderer
            .create(<PhotoViewer src={ImageUrls[0]}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
        

    } );
    }); 
    
    describe("selected image", () => {
        it("when I click the thumbnail the PhotoViewer updates with the new image", async () => {
            const component = render(<App/>);
            fireEvent.click(component.getAllByRole("img")[2]);
            const photoViewerURL=component.getAllByRole("img")[0].src;
            const selectedPhotoURL=component.getAllByRole("img")[2].src;
 
            expect(photoViewerURL).toBe(selectedPhotoURL);
        
        });

    });