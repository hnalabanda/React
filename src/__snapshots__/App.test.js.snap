// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selected image The photoviewer component has not changed 1`] = `
<div
  className="ViewerContainer"
>
  <img
    className="ViewerSize"
    src="https://i.picsum.photos/id/600/1600/900.jpg"
  />
</div>
`;
